stdlib/test/memory/test_maybe_uninitialized.mojo

```diff
@@ -89,7 +89,7 @@ def test_maybe_uninitialized_move_from_pointer():
     _ = a^
 
     # a is uninitialized now. Thankfully, we're working with trivial types
-    assert_equal(b.assume_initialized().move_count, 1)
+    assert_equal(b.assume_initialized().move_count, 2)
```

Comment:
This is quite strange indeed. Can you add a unit test with a non-movable type and see if it works correctly?

Reply:
Once I figure out stdlib dev with magic, i'll add it.

Reply:
@helehex did you dig into this a bit more?  Seems like a regression and/or a compiler bug.  If it's a bug, I'm sure @lattner would like to hear about it.

Reply:
The reason the move count goes up by one is because I'm doing two moves instead of one now. The actual moves should get optimized away, but it cant optimize the counter increment because that would be incorrect. (i'll explore cleaning it up some more though and getting rid of the extra move)

Reply:
Oh
Now it might be because it goes through `take_pointee()` first, then gets moved into the destination variable.

Reply:
Oh yeah, a non-movable test... i should try that

Reply:
Wait, if it's non-movable then you cant use `move_from`... could you maybe clarify what the test should look like?

Reply:
@JoeLoser Maybe `UnsafePointer.take_pointee()` should be using a named return slot?